extensions/mssql/src/objectExplorerNodeProvider/objectExplorerNodeProvider.ts

```diff
@@ -107,16 +108,31 @@ export class MssqlObjectExplorerNodeProvider extends ProviderBase implements azd
 			if (node) {
 				expandResult.errorMessage = node.getNodeInfo().errorMessage;
 				let children = await node.getChildren(true);
```

Comment:
As mentioned offline, we can add information on whether child expansion errored as a property of the nodes, and save the actual error object in that case. This would let us do:
```suggestion
				let children = await node.getChildren(true);
				if (node.expansionFailed && node.expansionError && node.expansionError.statusCode === 401) {
				    // prompt for credentials and re-expand
				}
```